src/nvim/ex_docmd.c

```diff
@@ -927,15 +927,15 @@ int do_cmdline(char_u *cmdline, LineGetter fgetline,
     if (retval == FAIL
         || (did_endif && KeyTyped && !did_emsg)
         ) {
-      need_wait_return = FALSE;
-      msg_didany = FALSE;               /* don't wait when restarting edit */
+      need_wait_return = false;
+      msg_didany = false;               // don't wait when restarting edit
     } else if (need_wait_return) {
       /*
        * The msg_start() above clears msg_didout. The wait_return we do
        * here should not overwrite the command that may be shown before
        * doing that.
        */
-      msg_didout |= msg_didout_before_start;
+      msg_didout = msg_didout || msg_didout_before_start;
```

Comment:
@bfredl Is this a neovim style thing? Is there a reason to prefer `x = x || y` over `x |= y`?

Reply:
not what I know of. the former should be fine, even for `bool`.